### DISCUSSION TITLE
Is it possible to use formatters, rust analyzer, etc... inside an injected section

### DISCUSSION THREAD

**Discussion by alanwsmith** [tree-sitter/tree-sitter#2993] [Q&A] [answered]
I built a tree-sitter parser for a custom format I'm working on. I've got it set up to highlight code sections properly. I've also got it set up in Neovim so I can hit a hotkey and it'll run the code and output the results into the file (like Org-Mode in emacs). What I haven't been able to figure out is if there's a way to have tools like Rust Analyzer, Prettier, etc... work while I'm inside the sections. Are there ways to do that?

Thanks!

-a

**Comment by amaanq** [comment, marked as answer]
Hey, sorry for the late response. I think you'd have to do this on top of tree-sitter, this is not something tree-sitter supports natively (code formatting), but it is 100% possible for you to write your own code/abstractions on top of tree-sitter in Neovim or wherever else to make this work.
